src/matter/cluster/server/OnOffServer.ts

```diff
@@ -0,0 +1,160 @@
+    //      In fact I need to get globalSceneControl
+    on: async ({ attributes: { onOff, onTime, offWaitTime, globalSceneControl } }) => {
+        onOff.set(true);
+        globalSceneControl.set(true);
+        if (onTime.get() === 0) {
+            offWaitTime.set(0);
+        }
+    },
+    off: async ({ attributes: { onOff, onTime, offWaitTime } }) => {
+        onOff.set(false);
+        onTime.set(0);
+    },
+    toggle: async ({ attributes: { onOff, onTime, offWaitTime } }) => {
+        if (onOff.get()) {
+            onOff.set(false);
+            onTime.set(0);
+        } else {
+            onOff.set(true);
+            if (onTime.get() === 0) {
+                offWaitTime.set(0);
+            }
+        }
+    },
+    offWithEffect: async ({request: { effectId, effectVariant}, attributes: { onOff, globalSceneControl } }) => {
+        // TODO implement logic with globalSceneControl, but before:
+        // globalSceneControl controls if "the server SHALL store its settings in its global scene" ... we need to add that first
+
+        if (globalSceneControl.get()) {
+            // TODO store settings in global scene
+            globalSceneControl.set(false);
+        }
+        // The global scene is defined as the scene that is stored with group identifier 0 and scene identifier 0.
+
+        // Just example implementation for now, I would more assume because we do not support dimming that
+        // we should just set off directly without delay?
+        // TODO In my eyes we need to access the "Level cluster" on the same endpoint to control the fading/dimming and such
+        switch (effectId) {
+            case OnOffEffectIdentifier.DelayedAllOff:
+                switch (effectVariant) {
+                    case OnOffDelayedAllOffEffectVariant.FadeToOffIn_0p8Seconds:
+                        setTimeout(() => onOff.set(false), 800);
+                        break;
+                    case OnOffDelayedAllOffEffectVariant.NoFade:
+                        onOff.set(false);
+                        break;
+                    case OnOffDelayedAllOffEffectVariant["50PercentDimDownIn_0p8SecondsThenFadeToOffIn_12Seconds"]:
+                        setTimeout(() => onOff.set(false), 12800);
+                        break;
+                }
+                break;
+            case OnOffEffectIdentifier.DyingLight:
+                switch (effectVariant) {
+                    case OnOffDyingLightEffectVariant["20PercentDimUpIn_0p5SecondsThenFadeToOffIn_1Second"]:
+                        setTimeout(() => onOff.set(false), 1500);
+                        break;
+                }
+                break;
+        }
+    },
+    onWithRecallGlobalScene: async ({ attributes: { onOff, globalSceneControl } }) => {
+        if (globalSceneControl.get()) {
+            return; // discard command
+        }
+        // TODO the Scene cluster server on the same endpoint SHALL recall its global scene, updating the OnOff
```

Comment:
Needs acces to scene cluster on same Endpoint (as soon as we implement it)

Reply:
Sounds good